### PR TITLE
Tidy namespaces and add some additional tests

### PR DESCRIPTION
**Pull request by magicweasel**
The namspaces are now not complaining ...

image

No disruption to the launch naming ...

image

I moved the test server to the directory to reflect the namespace ...

image

I didn't move `KestrelMock` to `KestrelMockServer` as was worried it would disrupt the build.

Also - added a handful of unit tests over some of my recent work (one of the corrections I recently made that had previously had an unhandled exception).

### CODE REVIEW

File: KestrelMock.Tests/MockServiceTest.cs [hunk 1]
```diff
@@ -4,22 +4,21 @@
 using Microsoft.AspNetCore.Mvc.Testing;
 using Microsoft.Extensions.Configuration;
 
-namespace KestrelMockServer.Tests
-{
-    public class MockTestApplicationFactory
+namespace KestrelMock.Tests;
```
Comment (magicweasel) [inline KestrelMock.Tests/MockServiceTest.cs:7]: Maybe you'd prefer `KestrelMockServer.Tests`?  The main reason for the change was to bring them line with each other (project and namespace match) ... so we could go the other way?

File: KestrelMock.Tests/DynamicMocks/response_after_dynamic_mock_creation___given_null_settings.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace KestrelMock.Tests.DynamicMocks;
+
+public class response_after_dynamic_mock_creation___given_null_settings
+{
+    private readonly DynamicMockAddedResponse response;
+
+    public response_after_dynamic_mock_creation___given_null_settings()
+    {
+        this.response = DynamicMockAddedResponse.Create(null);
```
Comment (magicweasel) [inline KestrelMock.Tests/DynamicMocks/response_after_dynamic_mock_creation___given_null_settings.cs:13]: This was one of the unhandled issues the other day ... but added a few tests over the rest of the behaviour in this area.

File: KestrelMock.Tests/MockServiceTest.cs (hunk 1 above)
Comment (JasonRowe) [reply KestrelMock.Tests/MockServiceTest.cs:7]: Looks more clear to me as KestrelMock.Tests

File: KestrelMock/KestrelMockServer.csproj
Comment (JasonRowe) [inline KestrelMock/KestrelMockServer.csproj:1]: Hmm might be odd the folder name and project name don't match now but we can revisit

**Review by JasonRowe — APPROVED** [review]

**Pull request merged**